File: client/src/app/types/device.ts
export interface Device {
    id?: string;
    capabilities?: string[];
    state?: DeviceState;
    light?: DeviceLight;
    colorXy?: DeviceColorXy;
    temperature?: DeviceTemperature;
    humidity?: DeviceHumidity;
    energy?: DeviceEnergy;
    contact?: DeviceContact;
}

export interface DeviceState {
    state: boolean;
}

export interface DeviceLight {
    brightness: number;
}

export interface DeviceColorXy {
    x: number;
    y: number;
}

export interface DeviceTemperature {
    value: number;
    unit: string;
}

export interface DeviceHumidity {
    value: number;
    unit: string;
}

export interface DeviceEnergy {
    total?: number;
    current?: number;
    voltage?: number;
    power?: number;
}

export interface DeviceContact {
    value?: boolean;
}

File: server/SOC.IoT.Domain/Entity/Device.cs
using SOC.IoT.Domain.Enum;
using System.ComponentModel.DataAnnotations;

namespace SOC.IoT.Domain.Entity;

public class Device
{
    [Required]
    public string Id { get; set; }
    public ICollection<DeviceCapability> Capabilities { get; set; } = new List<DeviceCapability>();

    public DeviceState? State { get; set; }
    public DeviceLight? Light { get; set; }
    public DeviceColorXy? ColorXy { get; set; }
    public DeviceTemperature? Temperature { get; set; }
    public DeviceHumidity? Humidity { get; set; }
    public DeviceEnergy? Energy { get; set; }
    public DeviceContact? Contact { get; set; }
}
